### DISCUSSION TITLE
Check for missing files?

### DISCUSSION THREAD

**Discussion by edent** [beetbox/beets#4738] [Q&A] [answered]
I have a library with 12,000 tracks (!) - is there any way to check the library to make sure that they're all present on the filesystem?

Is there any way to prune unexpected paths / files from the library's directory?

This question inspired by the way the Calibre book library can check the integrity of its library.

(I've seen the "check" plugin - but that's about checking the file itself hasn't changed. I want to make sure I haven't accidentally deleted an album.)

**Comment by jackwilsdon** [comment, marked as answer]
The [badfiles](https://docs.beets.io/en/latest/plugins/badfiles.html) plugin sounds like what you're looking for.

**Reply by edent** [reply]
Thanks - that's helpful.  I've confirmed that it will show when a file is missing. But, blimey, it's a chatterbox!

**Reply by xeruf** [reply]
how can I prune nonexistent files from the library?

**Reply by ThinkChaos** [reply]
@xeruf @psyburr Was also looking for that and found it's `beet update` (`-p`/`--pretend`) to check first

https://beets.readthedocs.io/en/stable/reference/cli.html#update
https://beets-users.narkive.com/VV9mHDXH/beets-finding-and-removing-missing-files-not-tracks
